src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,104 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int currentCapacity;
+    private int size;
+    private int threshold;
 
     @Override
     public void put(K key, V value) {
-
+        if (++size >= threshold || currentCapacity == 0) {
+            resize();
+        }
+        putValue(hash(key), key, value);
     }
 
     @Override
     public V getValue(K key) {
+        if (table == null) {
+            return null;
+        }
+        int hash = hash(key);
+        Node<K, V> currentNode = table[hash];
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value) {
+            this.key = key;
+            this.value = value;
+            this.hash = hash;
+        }
+    }
+
+    private void resize() {
+        Node<K, V>[] oldTable = table;
+        int oldCapacity = table == null ? 0 : table.length;
```

Comment:
just initialize table in empty constructor, and you can remove all these null checks